app/src/main/java/com/eveningoutpost/dexdrip/webservices/WebServiceSgv.java

```diff
@@ -100,23 +110,58 @@ public WebResponse request(String query) {
             brief = true;
         }
 
-
         final JSONArray reply = new JSONArray();
 
         // whether to include data which doesn't match the current sensor
         final boolean ignore_sensor = Home.get_follower() || cgi.containsKey("all_data");
 
-        final List<BgReading> readings = BgReading.latest(count, ignore_sensor);
+        // Store a cache of the last BgReading.latest() query for the duration in which there is no
+        // new latest reading. Since obtaining the latest reading is fast, but a larger number of
+        // readings is significantly slower, this optimizes the most often use case.
+        List<BgReading> bgr = BgReading.latest(1, ignore_sensor);
+        BgReading latestReading = null;
+        if (bgr != null && bgr.size() > 0) {
+            latestReading = bgr.iterator().next();
+        }
+
+        List<BgReading> readings;
+        if (cachedReadings != null && cachedReadings.size() > 0 && count <= cachedReadings.size() &&
+            latestReading != null && latestReading.uuid.equals(cachedReadings.iterator().next().uuid))
+        {
```

Comment:
The `{` could be at the end of the previous line. :-o

Reply:
Done